eZ/Publish/Core/Repository/ContentTypeService.php

```diff
@@ -1639,4 +1640,14 @@ public function removeContentTypeTranslation(APIContentTypeDraft $contentTypeDra
 
         return $this->contentTypeDomainMapper->buildContentTypeDraftDomainObject($contentType);
     }
+
+    public function deleteUserDrafts(int $userId): void
+    {
+        $params = new SPIContentType\DeleteByParamsStruct([
+            'modifierId' => $userId,
+            'status' => ContentType::STATUS_DRAFT,
+        ]);
+
+        $this->contentTypeHandler->deleteByParams($params);
```

Comment:
I don't like `deleteByParams`. Method doesn't reveal what it deletes and this is raises a single-responsibility issues for me. Meaning that with a growth of DeleteByParamsStruct, more responsibility will be given to `Handler::deleteByParams` instead of creating dedicated methods which are explicit about their operation.

There's nothing wrong with `Handler::deleteUserDraftsWithStatus(int $userId, $status)`. It's an implementation detail that _should_ be exposed at this point.

POV ping @Nattfarinn @webhdx

Reply:
Remark no. 2:

It's not safe to just delete regardless of system state.
You should check first if user does not exist and proceed only if it does.

Optional: If it does exist, then at least `class/delete` should be checked.

Reply:
I prefer to have `deleteByParams` rather then `deleteByX` , `deleteByY`, `deleteByZ` and so on and Handler with hundreds of lines of code. It is just less mess.

Reply:
Would that just not moved that extra few lines into additional if statements inside `Content/Type/Handler` and `Cache/ContentTypeHandler`?

Reply:
> I prefer to have `deleteByParams` rather then `deleteByX` , `deleteByY`, `deleteByZ` and so on and Handler with hundreds of lines of code. It is just less mess.

So instead you'll end-up with a spaghetti of ifs inside one method. That is really a contradiction to SOLID.

Reply:
> Would that just not moved that extra few lines into additional if statements inside `Content/Type/Handler` and `Cache/ContentTypeHandler`?

It is really hard to answer this question. Its depends on what kind of programmer will be implementing this. If at all.

Reply:
> > I prefer to have `deleteByParams` rather then `deleteByX` , `deleteByY`, `deleteByZ` and so on and Handler with hundreds of lines of code. It is just less mess.
> 
> So instead you'll end-up with a spaghetti of ifs inside one method. That is really a contradiction to SOLID.

Why are you considering implementing this with spaghetti code and lots of ifs? I agree that it would be against SOLID.

Reply:
> Why are you considering implementing this with spaghetti code and lots of ifs? I agree that it would be against SOLID.

Adding properties in the future to struct requires adding new ifs to `deleteByParams`. This is not SOLID.

Reply:
I couldn't agree that it `requires`.

Reply:
I strongly agree with Andrew on this one. This approach will be a mess in the future as it can grow and easily get out of hand. We need dedicated methods as we always do.

Reply:
This method gives no contract to what it does. You have to create separate methods to keep contract on every possible operation. Handler is an interface, how would you ensure same capabilities in classes that implement it if the logic depends on whatever you put into an array?

Reply:
@pawbuj please adjust it accordingly to comments, valid points here

Reply:
> I strongly agree with Andrew on this one. This approach will be a mess in the future as it can grow and easily get out of hand. We need dedicated methods as we always do.

imo useless argument, every piece of code can `can grow and easily get out of hand`